### PR TITLE
[CRM-53] Add new customer logic and tests

### PR DESCRIPTION
**Pull request by aliwocha**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Commits are suqashed
- [x] PR and commit messages follow the style guidelines

### CODE REVIEW

File: backend/src/main/java/com/crm/dto/request/CustomerRequest.java [hunk 1]
```diff
@@ -25,8 +25,6 @@
     // Allowing polish numbers only without prefixes, dashes or white spaces
     private static final String PHONE_PATTERN = "^(45|5[0,1,3,7]|6[0,6,9]|7[2,3,8,9]|88)[0-9]{7}";
```
Comment (SteelWro) [inline backend/src/main/java/com/crm/dto/request/CustomerRequest.java:26]: Add '$' character at the end of pattern. It matches the end of a line

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ void shouldReturnCorrectResponseStatusWhenCallingCarWithExistingId_WhenDescripti
     }
 
     @Test
-    void shouldReturnCorrectResponseStatusWhenCallingCarWithExistingId_WhenDescriptionIsBlank() throws Exception {
+    void shouldResponseWith_OK_WhenCallingCarWithExistingId_WhenDescriptionIsBlank() throws Exception {
         mvc.perform(get("/cars/4"))
```
Comment (SteelWro) [inline backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:85]: Maybe add
.andExpect(status().is(OK.value()))

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java [hunk 3]
```diff
@@ -98,7 +98,7 @@ void shouldReturnCorrectResponseStatusWhenCallingCarWithExistingId_WhenDescripti
     }
 
     @Test
-    void shouldReturnCorrectResponseStatusWhenCallingCarWithExistingId_WhenDescriptionIsNull() throws Exception {
+    void shouldResponseWith_OK_WhenCallingCarWithExistingId_WhenDescriptionIsNull() throws Exception {
         mvc.perform(get("/cars/6"))
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON))
```
Comment (SteelWro) [inline backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:103]: Maybe add
.andExpect(status().is(OK.value()))

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java [hunk 4]
```diff
@@ -115,13 +115,13 @@ void shouldReturnCorrectResponseStatusWhenCallingCarWithExistingId_WhenDescripti
     }
 
     @Test
-    void shouldReturnBadResponseStatusWhenCallingCarWithNonExistingId() throws Exception {
+    void shouldResponseWith_NotFound_WhenCallingCarWithNonExistingId() throws Exception {
         mvc.perform(get("/cars/1000"))
                 .andExpect(status().is(NOT_FOUND.value()));
```
Comment (SteelWro) [inline backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:120]: Maybe add expect message

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java [hunk 5]
```diff
@@ -138,13 +138,13 @@ void shouldReturnCorrectResponseStatusWhenCallingCarWithValidRegistrationNumber(
...
+    void shouldResponseWith_NotFound_WhenCallingCarWithInvalidRegistrationNumber() throws Exception {
         mvc.perform(get("/cars/search?license-plate=KR"))
                 .andExpect(status().is(NOT_FOUND.value()));
     }
 
     @Test
-    void shouldReturnBadResponseStatusWhenCallingCarWithBlankRegistrationNumber() throws Exception {
+    void shouldResponseWith_NotFound_WhenCallingCarWithBlankRegistrationNumber() throws Exception {
         mvc.perform(get("/cars/search?license-plate="))
                 .andExpect(status().is(NOT_FOUND.value()));
```
Comment (SteelWro) [inline backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:149]: Maybe add expect message

File: backend/src/main/java/com/crm/service/impl/CustomerServiceImpl.java [hunk 6]
```diff
@@ -15,33 +17,57 @@
...
+
+        CustomerEntity newCustomer = customerMapper.convertToEntity(customerRequest);
+        CustomerEntity savedCustomer = customerRepository.save(newCustomer);
+
+        return customerMapper.convertToDto(savedCustomer);
+    }
+
+    private boolean isDuplicateCustomer(CustomerRequest customerRequest, CustomerEntity c) {
+        return c.getName().equals(customerRequest.getName()) && c.getSurname().equals(customerRequest.getSurname())
+                && c.getAddress().equals(customerRequest.getAddress());
```
Comment (SteelWro) [inline backend/src/main/java/com/crm/service/impl/CustomerServiceImpl.java:71]: Troche lipa z tym adresem, może z adresu też zrobić encję?
-country
-city
-postal code
-street name
-street number
-flat number

do tego equals and hashcode

Aktualnie jeśli cały adres jest w Stringu to trzeba pilnować zapisu całego adresu żeby equals przechodził

Comment (aliwocha) [reply backend/src/main/java/com/crm/service/impl/CustomerServiceImpl.java:71]: To zadanie już jest w planach, właściwie to ja je sobie przypisałem i chcę to zrobić jak tylko znajdę czas :)

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java (hunk 5 above)
Comment (aliwocha) [reply backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:149]: Done

File: backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java (hunk 4 above)
Comment (aliwocha) [reply backend/src/test-integration/java/com/crm/api/v1/CarControllerIT.java:120]: Done

**Pull request merged**